MLS.Agent.Tools/TypeExtensions.cs

```diff
@@ -16,5 +16,13 @@ public static string ReadManifestResource(this Type type, string resourceName)
                 return reader.ReadToEnd();
             }
         }
+
+        public static bool CanBeInstantiatedFrom(this Type type, Type resultantType)
```

Comment:
This can be decomposed a little. Assignability is a different concern from instantiability so should probably not be included in the method.